modules/plugins/statusline/feline/feline.nix

```diff
@@ -0,0 +1,144 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  inherit (builtins) elem;
+  inherit (lib.options) mkOption mkEnableOption literalExpression;
+  inherit (lib.types) str listOf attrs attrsOf anything either submodule;
+  inherit (lib.lists) optional;
+  inherit (lib.generators) mkLuaInline;
+  inherit (lib.nvim.types) mkPluginSetupOption;
+
+  conditionalRenderers = {
+    options = {
+      filetypes = mkOption {
+        type = listOf str;
+        default = [
+          "^NvimTree$"
+          "^neo-tree$"
+          "^startify$"
+          "^fugitive$"
+          "^fugitiveblame$"
+          "^qf$"
+          "^help$"
+        ];
+
+        description = "Filetypes in which to force render inactive statusline";
+      };
+
+      buftypes = mkOption {
+        type = listOf str;
+        default = ["^terminal$"];
+        description = "Buffer types in which to force render inactive statusline";
+      };
+
+      bufnames = mkOption {
+        type = listOf str;
+        default = [];
+        description = "Buffer names in which to force render inactive statusline";
+      };
+    };
+  };
+in {
+  options.vim.statusline.feline-nvim = {
+    enable = mkEnableOption "minimal, stylish and customizable statusline, statuscolumn, and winbar [feline.nvim]";
+    setupOpts = mkPluginSetupOption "feline-nvim" {
+      custom_providers = mkOption {
+        type = attrsOf anything;
```

Comment:
Should be of type luaInline

Reply:
`custom_providers` takes a table, in which case we should be able to get a `listOf luaInline`. No?

Reply:
Men go read the docs, you have to be able to the custom providers somehow. It's a set